app/gen-server/lib/HomeDBManager.ts

```diff
@@ -1952,14 +1952,22 @@ export class HomeDBManager extends EventEmitter {
 
   // Update the webhook url in the webhook's corresponding secret (note: the webhook identifier is
   // its secret identifier).
-  public async updateWebhookUrl(id: string, docId: string, url: string, outerManager?: EntityManager) {
+  public async updateWebhookUrlAndAuth(
+    id: string, docId: string, url: string | undefined, auth: string | undefined, outerManager?: EntityManager) {
     return await this._runInTransaction(outerManager, async manager => {
       const value = await this.getSecret(id, docId, manager);
       if (!value) {
         throw new ApiError('Webhook with given id not found', 404);
       }
       const webhookSecret = JSON.parse(value);
-      webhookSecret.url = url;
+      // As we want to patch the webhookSecret object, only set the url and the authorization when they are defined.
+      // When the user wants to empty the value, we are expected to receive empty strings.
+      if (url !== undefined) {
```

Comment:
2 things:
- I'd convert arguments to an object, as there are too many strings there
- I'd throw an exception when both those things are not defined.

If you take a look at the line 1939 `updateSecret` method, it will throw an exception if the update won't do anything, but with completely misleading error message - that the secret wasn't found, when in fact it was found but just not updated.

Btw: can you update that error message as well, to something like:
`secret with given id not found or nothing was updated`.